Core/Resource/SymbolJson.cs

```diff
@@ -368,9 +336,144 @@ public Symbol ReadSymbol(Model model, JToken o, bool allowNonOperatorInstanceTyp
+            Application.Exit();
+            Application.ExitThread();
+
+            return null;
+        }
+        
+        // Method for when allowNonOpInstanceType = false
+        static Type GetAnyInstanceType(string typeName)
+        {
+            var typeIsCached = TypeCache.TryGetValue(typeName, out var type);
+            if (typeIsCached)
+            {
+                return type ?? ObjectType;
+            }
+
+            type = Type.GetType(typeName) ?? ObjectType;
+            TypeCache.TryAdd(typeName, type);
+
+            return type;
         }
+        
+        private static void ReadConnections(JArray connectionsJson, string name, HashSet<Guid> missingSymbolChildIds, List<Symbol.Connection> connections)
+        {
+            foreach (var c in connectionsJson)
+            {
+                var connection = ReadConnection(c);
+
+                var undefinedType = missingSymbolChildIds.Contains(connection.TargetParentOrChildId)
+                                    || missingSymbolChildIds.Contains(connection.SourceParentOrChildId);
+
+                if (undefinedType)
+                {
+                    Log.Warning($"Skipping connection to child of undefined type in {name}");
+                    continue;
+                }
+
+                connections.Add(connection);
+            }
+        }
+
+        private static readonly Type ObjectType = typeof(object);
+        // Define a static dictionary to cache Type instances by instanceTypeName
+        private static readonly ConcurrentDictionary<string, Type> TypeCache = new();
         #endregion
+        
+        internal readonly struct JsonKeys
+        {
+            public const string Connections = "Connections";
+            public const string SourceParentOrChildId = "SourceParentOrChildId";
+            public const string SourceSlotId = "SourceSlotId";
+            public const string TargetParentOrChildId = "TargetParentOrChildId";
+            public const string TargetSlotId = "TargetSlotId";
+            public const string Children = "Children";
+            public const string Id = "Id";
+            public const string SymbolId = "SymbolId";
+            public const string Name = "Name";
+            public const string Namespace = "Namespace";
+            public const string InputValues = "InputValues";
+            public const string OutputData = "OutputData";
+            public const string Type = "Type";
+            public const string DirtyFlagTrigger = "DirtyFlagTrigger";
+            public const string IsDisabled = "IsDisabled";
+            public const string DefaultValue = "DefaultValue";
+            public const string Value = "Value";
+            public const string Inputs = "Inputs";
+            public const string Outputs = "Outputs";
+            public const string Animator = "Animator";
+        }

```

Comment:
Might not be the prettiest way to have a whole bunch of constant strings, but I couldn't think of a prettier way